sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/SpringAppImpl.java

```diff
@@ -331,11 +335,10 @@ public SpringAppImpl withDefaultActiveDeployment() {
     public <T extends
         SpringAppDeployment.DefinitionStages.WithAttach<? extends SpringApp.DefinitionStages.WithCreate, T>>
         SpringAppDeployment.DefinitionStages.Blank<T> defineActiveDeployment(String name) {
-        return (SpringAppDeployment.DefinitionStages.Blank<T>) deployments.define(name);
+        return (SpringAppDeployment.DefinitionStages.Blank<T>) deployments.define(name).withActivation();
     }
 
-    SpringAppImpl addActiveDeployment(SpringAppDeploymentImpl deployment) {
-        withActiveDeployment(deployment.name());
+    SpringAppImpl addDeployment(SpringAppDeploymentImpl deployment) {
```

Comment:
Since the define method is called `defineActiveDeployment`, I added `withActivation()` in its implementation.

`addActiveDeployment` is called by `attach` to the SpringApp. Clearly `addDeployment` is a better name, and `withActivation` shouldn't be done here.